xbmc/interfaces/json-rpc/FileOperations.cpp

```diff
@@ -213,6 +214,50 @@ JSONRPC_STATUS CFileOperations::GetFileDetails(const std::string &method, ITrans
   return OK;
 }
 
+JSONRPC_STATUS CFileOperations::SetFileDetails(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  std::string file = parameterObject["file"].asString();
+  if (!CFile::Exists(file))
+    return InvalidParams;
+
+  if (!CFileUtils::RemoteAccessAllowed(file))
+    return InvalidParams;
+
+  std::string media = parameterObject["media"].asString();
+  StringUtils::ToLower(media);
+
+  if (media.compare("video") != 0)
+    return InvalidParams;
+
+  CVideoDatabase videodatabase;
+  if (!videodatabase.Open())
+    return InternalError;
+
+  int fileId = videodatabase.AddFile(file);
+
+  CVideoInfoTag infos;
+  if (!videodatabase.GetFileInfo("", infos, fileId))
+    return InvalidParams;
+
+  CDateTime lastPlayed = infos.m_lastPlayed;
+  int playcount = infos.m_playCount;
+  if (!parameterObject["lastplayed"].isNull())
+  {
+    lastPlayed.Reset();
+    SetFromDBDateTime(parameterObject["lastplayed"], lastPlayed);
+    playcount = lastPlayed.IsValid() ? 1 : 0;
```

Comment:
Do we really want this? I would expect the following behaviour:
* If lastplayed is set to an invalid value (e.g. empty string) then playcount becomes `0`.
* If lastplayed is set to a valid value then playcount either becomes `1` or stays the existing value if it's already bigger than or equal to `1`.